app/javascript/components/catalog-form/catalog-form.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import createSchema from './catalog-form.schema';
+import { filterOptions } from '../dual-list-select/helpers';
+import { API } from '../../http_api';
+import { cleanVirtualDom } from '../../miq-component/helpers';
+
+class CatalogForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      schema: createSchema([]),
+      initialValues: {
+        name: undefined,
+        description: undefined,
+        service_templates: [],
+      },
+      isLoaded: false,
+    };
+    cleanVirtualDom();
+  }
+
+  componentDidMount() {
+    miqSparkleOn();
+    const { catalogId } = this.props;
+    if (catalogId) {
+      Promise.all([
+        API.get('/api/service_templates?expand=resources&filter[]=service_template_catalog_id=null'),
+        API.get(`/api/service_catalogs/${catalogId}?expand=service_templates`)])
+        .then(([{ resources }, { name, description, service_templates }]) => {
+          const editedDescription = description.length > 0 ? description : undefined;
```

Comment:
You can just chech if description === null no need for length checking

Reply:
Thanks! Wasn't sure if null and '' is the same. :smile: